data_source/gateway.go

```diff
@@ -27,14 +27,153 @@ func (g *Gateway) GetBlockByNumber(blockNumber uint64) (*core.Block, error) {
 
 // GetTransaction gets the transaction for a given transaction hash from the feeder gateway,
 // then adapts it to the appropriate core.Transaction types.
-func (g *Gateway) GetTransaction(transactionHash *felt.Felt) (*core.Transaction, error) {
-	return nil, errors.New("not implemented")
+func (g *Gateway) GetTransaction(transactionHash *felt.Felt) (core.Transaction, error) {
```

Comment:
```suggestion
func (g *Gateway) GetTransaction(transactionHash *felt.Felt) (*core.Transaction, error) {
```

The above should fix the error